packages/goblint-cil/goblint-cil.1.8.0/opam

```diff
@@ -0,0 +1,67 @@
+opam-version: "2.0"
+synopsis:
+  "A front-end for the C programming language that facilitates program analysis and transformation"
+description: """\
+This is a fork of the 'cil' package needed to build 'goblint'.
+Changes:
+- Proper support for C99, (#9) and VLAs in particular (#5, #7)
+- It uses Zarith instead of the deprecated Num
+- Support for more recent OCaml versions (≥ 4.06)
+- Large integer constants that do not fit in an OCaml int are represented as a string instead of getting truncated
+- Syntactic search extension (#21)
+- Some warnings were made optional
+- Unmaintained extensions (#30) were removed
+- Many bug fixes"""
+maintainer: [
+  "Michael Schwarz <michael.schwarz93@gmail.com>"
+  "Ralf Vogler <ralf.vogler@gmail.com>"
+]
+authors: "gabriel@kerneis.info"
+license: "BSD-3-Clause"
+homepage: "https://cil-project.github.io/cil/"
+bug-reports: "https://github.com/goblint/cil/issues"
+depends: [
+  "ocaml" {>= "4.04.2"}
+  "ocamlfind"
+  "zarith"
+  "hevea" {with-doc}
+  "dune" {>= "2.7" & >= "2.7"}
+  "odoc" {with-doc}
+  "stdlib-shims"
+  "ppx_deriving_yojson"
+  "yojson"
+  "batteries" {>= "3.2.0"}
+  "conf-perl"
+]
+conflicts: ["cil"]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+depexts: [
+  ["libc6-dev-i386"] {os-family = "debian" | os-family = "ubuntu"}
+  ["glibc-devel-32bit"] {os-family = "suse"}
+  ["glibc-devel.i686"]
+    {os-distribution = "fedora" | os-distribution = "rhel" |
+     os-distribution = "centos" |
+     os-distribution = "ol"}
+  ["lib32-glibc-devel"] {os-family = "arch"}
```

Comment:
I'm curious, why is this required? If you require the user to use x86_32 you can simply say so by replacing this by:
```
available: arch = "x86_32"
```
Though I would advise against it and would recommend a more portable approach (which would be neither requiring x86_32 glibc or requiring the package to be installed on x86_32 only

Reply:
These packages are the 32bit headers of glibc for a 64bit architecture. `goblint-cil` attempts to build a [machine model](https://people.eecs.berkeley.edu/~necula/cil/cil007.html) also for the case where one is compiling for a 32bit target during the testing phase. However, the tests still succeed if this x86 machine model can not be built, so I will remove this requirement again.